src/utils.c

```diff
@@ -170,7 +170,7 @@ ts_interval_value_to_internal(Datum time_val, Oid type_oid)
 			if (interval->month != 0)
 				ereport(ERROR,
 						(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						 errmsg("intervals must be defined in terms of days or smaller")));
+						 errmsg("intervals must be defined as a fixed duration interval (such as weeks, days, hours, minutes, seconds, etc.)")));
 			return interval->time + (interval->day * USECS_PER_DAY);
```

Comment:
Not entirely sure what "fixed duration" means in this context. Isn't a month a fixed duration?

If I understand this check correctly, it only checks that the duration is not one month (or more) because the number of months is ignored in the computation last.

The new message suggest that `INTERVAL '1 MONTH'`would be OK, but it will still generate an error.

(And the old message suggested that `INTERVAL '1 WEEK'` would not be OK, but it would not generate an error, so the message is not really accurate.)

Reply:
Hrm.  Month isn't a fixed duration here because it changes based /which/ month (28, 30, or 31 days) or even if it's a leap year.  That's the main reason `time_bucket` doesn't allow you to bucket by months...

Reply:
I suggest being short and explicit: `"months and years not supported"` for the main error message. Then add the longer message in `errdetail` (or hint) instead: `"an interval must be defined as a fixed duration (such as weeks, days, hours, minutes, seconds, etc.)"`

Reply:
@erimatnor Updated to your suggestion.